eLCI flow mapping update

frankhanna94 opened the pull request:
Context of this pull request: the eLCI.csv mapping document and flow mapping function from elementaryflows.py were used in 'https://github.com/NETL-RIC/ElectricityLCI' to develop a function that directly generates the emissions inventory for natural gas production through transmission using the NG model published on NETL's Energy Data eXchange (EDX).

Proposed changes: To complete the flow mapping exercise, we propose 68 new emissions to be added to the flow map.

Note: Despite this change, three flows remained unmapped: 1) Acidity, 2) Total Dissolved Solids 3) Total non-methane organic carbon

frankhanna94 committed:
eLCI flowmapping update

frankhanna94 committed:
UTF-8 encoding

bl-young commented on fedelemflowlist/flowmapping/eLCI.csv:
```diff
@@ -4282,7 +4282,7 @@ eLCI,Black carbon,,emission/air,kg,=,1,Carbon black,34e3e9d7-a951-3525-a73b-cfb3
 eLCI,"Dichlorobenzene (m-DCB; 1,3-Dichlorobenzene)",,emission/air,kg,=,1,m-Dichlorobenzene,54f426c2-5eb6-3d54-99e6-41cfa6d6e715,emission/air,kg,Hanna,Henriksen,12/5/2025
 eLCI,Ethylendichloride,,emission/air,kg,=,1,"1,2-Dichloroethane",c2d517e6-4c5d-3002-8a0d-54434c5a3bae,emission/air,kg,Hanna,Henriksen,12/5/2025
-eLCI,Hydrogen�sulphide ,,emission/air,kg,=,1,Hydrogen sulfide,f8cbcd43-8201-3b05-b19a-82d305414af4,emission/air,kg,Hanna,Henriksen,12/5/2025
+eLCI,Hydrogen�sulphide ,,emission/air,kg,=,1,Hydrogen sulfide,f8cbcd43-8201-3b05-b19a-82d305414af4,emission/air,kg,Hanna,Henriksen,12/5/2025
```
so is it encoded like this in your source file?

bl-young replied:
oh nm i see that's fixed now nad no longer causing an error

frankhanna94 replied:
Thank you for pointing that out. Actually, I was getting an error suggesting that the changes we made to the file changed its encoding. I saved the file locally with Encoding UTF-8 to get rid of the error
I think that swapped the space with these characters which might cause problems in its application. 
I will make another check!

frankhanna94 replied:
I made/committed the change directly here - I believe it should be all good now for the new additions.

frankhanna94 committed:
eLCI UUID Edit

frankhanna94 committed:
Fix typo in Hydrogen sulphide entry in eLCI.csv

bl-young commented on fedelemflowlist/flowmapping/eLCI.csv:
```diff
@@ -4224,10 +4224,78 @@ eLCI,Decane,,emission/water,kg,=,1,Decane,0c8c9877-5938-36dd-910b-48daed6913e1,e
 eLCI,"Toluene, vinyl",,emission/air,kg,=,1,Vinyltoluene,926e6e98-b7ed-38f3-9873-93aeaf5f33e6,emission/air,kg,Hottle,Edelen,2/5/2020
 eLCI,Chlorides,,emission/water,kg,<,1,"Chlorides and sulfates, total",1fd6ca1f-26bf-3d26-8170-55746d2e2522,emission/water,kg,Hottle,Young,2/10/2020
 eLCI,Tetrachloroethene,,emission/air,kg,=,1,Tetrachloroethane,5e5215d6-2374-38ee-b26d-3a128f2e6ecd,emission/air,kg,Hottle,Edelen,2/10/2020
+eLCI,Aluminium,,emission/ground,kg,=,1,Aluminum,ce7ca5a4-4642-3f68-b2e6-61853872844d,emission/ground,kg,Hanna,Henriksen,12/5/2025
```
i found 3 duplicates, line 4234 aluminum, line 4240 beryllium, and line 4277 1,2,3-trimethylbenzen.

i think that's what caused the second failure

frankhanna94 replied:
Agreed! I removed these duplicates.
I do recall adding those because for some reason, my code wasn't capturing the previous entries by Hottle/Jamieson (despite removing spaces and converting everything to lowercase). Not sure if the file encoding had something to do with it. I guess we'll test that in the future and figure something out if we still get that problem.

frankhanna94 committed:
Removed duplicate chemical entries in eLCI.csv

Removed duplicate entries for Aluminium, Beryllium, and 1,2,3-Trimethylbenzene.

bl-young commented on fedelemflowlist/flowmapping/eLCI.csv:
```diff
@@ -4224,10 +4224,75 @@ eLCI,Decane,,emission/water,kg,=,1,Decane,0c8c9877-5938-36dd-910b-48daed6913e1,e
+eLCI,Total dissolved organic carbon,,emission/water,kg,=,1,Total Organic Carbon,cbf6df8e-c4a2-3920-a97a-2238ccb7bc6c,emission/water,kg,Hanna,Henriksen,12/5/2025
+eLCI,Total Iron,,emission/water,kg,=,1,Iron,2b61d376-c8c2-31ae-9c19-9b8b958c74dd,emission/water,kg,Hanna,Henriksen,12/5/2025
+eLCI,Total Kjedahl Nitrogen,,emission/water,kg,=,1,Nitrogen,d4625608-6ef8-3ce6-bfa4-0f96f3535e23,emission/water,kg,Hanna,Henriksen,12/5/2025
+eLCI,Total organic carbon,,emission/water,kg,=,1,Nitrogen,cbf6df8e-c4a2-3920-a97a-2238ccb7bc6c,emission/water,kg,Hanna,Henriksen,12/5/2025
```
this mapping is wrong. UUID doesn't match Nitrogen. i assume the flow name is incorrect

frankhanna94 replied:
Yes that should be - total organic carbon - I changed it

bl-young commented on fedelemflowlist/flowmapping/eLCI.csv:
```diff
@@ -4224,10 +4224,75 @@ eLCI,Decane,,emission/water,kg,=,1,Decane,0c8c9877-5938-36dd-910b-48daed6913e1,e
+eLCI,Lithium,,emission/ground,kg,=,1,Lithium,d989d2b9-fa3f-3b64-805d-f4bb267c9a0e,emission/ground,kg,Hanna,Henriksen,12/5/2025
+eLCI,nitrate-nitrite,,emission/ground,kg,=,1,Nitrate,d6d40356-2c5c-3ae0-89a9-fec2a41f07b2,emission/ground,kg,Hanna,Henriksen,12/5/2025
+eLCI,Nitrite,,emission/ground,kg,=,1,Nitrite,ec1e5b2d-d3c6-31b7-9782-a5604248a90c,emission/ground,kg,Hanna,Henriksen,12/5/2025
+eLCI,Phenols,,emission/ground,kg,=,1,Phenols ,d0527c98-4239-332b-83f9-58ff59a84047,emission/ground,kg,Hanna,Henriksen,12/5/2025
```
extra space in target flow name causing an error here.

frankhanna94 replied:
Fixed!

bl-young reviewed (changes requested):
ok sorry my previous suggested fix did not solved our problem but I found the two issues

frankhanna94 committed:
Fix spelling of 'Phenols' and update 'Total organic carbon'

bl-young commented:
bypassing develop
